lib/multi_aggregator/data_transfer.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module MultiAggregator
+  class DataTransfer
+    attr_reader(
+      :logger
+    )
+
+    def initialize(
+      logger = MultiAggregator::Logger.new
+    )
+      @logger = logger
+    end
+
+    def call(query_spec, storage, providers = {})
+      query_spec.each do |provider_id, tables_spec|
+        transfer_tables(storage, providers[provider_id], provider_id, tables_spec)
```

Comment:
Why do you stick to data copy? You copy data and lose all the indexes and other optimizations. E.g. how would you copy elastic search data into PG? 

I would vote for more sophisticated query plan. Could you please suggest the basic workflow and couple of basic optimizations for this solution?